File: code/VideoRest.php
<?php
require_once(realpath(dirname(__FILE__)) . '/LiveChat.php');
require_once(realpath(dirname(__FILE__)) . '/User.php');
require_once(realpath(dirname(__FILE__)) . '/VideoStream.php');
require_once(realpath(dirname(__FILE__)) . '/Video.php');

use LiveChat;
use VideoStream;
use User;
use Video;

/**
 * @access public
 * @author mariaroxanaluca
 */
class VideoRest {
	/**
	 * @AttributeType String
	 */
	private $_meetingId;
	/**
	 * @AttributeType String
	 */
	private $_uRI;

	/**
	 * @AttributeType User
	 * /**
	 *  * @AssociationType User
	 *  * @AssociationMultiplicity 1
	 *  * /
	 */
	public $_VideoStream_;

	public $_Videos_ = array();
	/**
	 * @AttributeType Settings
	 * /**
	 *  * @AssociationType Settings
	 *  * @AssociationMultiplicity 1
	 *  * /
	 */

	private $_ViCoordinator_;

	public __construct( $ViCoordinator){
		$_ViCoordinator_ = $ViCoordinator;
		$meeting_id = random_bytes(20);
		$url_link = "";
		self::$_VideoStream_ = VideoStream();
	}

	public function getState() {
		 return $this->$_VideoStream_;
	}

	public function setSettings($user,$settings)
	{
		if ($user == $_ViCoordinator_){
			$_VideoStream_ = VideoStream($settings);
			notifyAllObservers();
		}
	}

	public function attach($user){
		 $observer = Video($user);
		 $_Videos_.add($observer);
	}

	public function notifyAllObservers(){
		 foreach ( $observer as $_Videos_) {
				$observer.update();
		 }
	}

	public function checkCredits($url_link, $meeting_id){
		return ($_uRI === $url_link && $_meetingId===$meeting_id);
	}

}
?>

File: backend/src/main/java/com/viSmart/viSmart/VideoRest.java
package com.viSmart.viSmart;

import javax.websocket.Session;
import java.util.ArrayList;
import java.util.Random;
import java.util.UUID;

public class VideoRest{

	private String meetingId;
	private String URI;

	private ArrayList<Video> Videos = new ArrayList<>();

	private User coordinator;
	private String data;
	private VideoStream videoStream;

	public VideoRest( User coord){
		Random random = new Random();
		coordinator = coord;
		meetingId = UUID.randomUUID().toString();
		URI = "";
	}

	public VideoStream getState() {
		return videoStream;
	}

	public void setSettings(User user,Settings settings,String data)
	{
		if (user.equals(coordinator)){
			videoStream = new VideoStream(settings);
			notifyAllObservers(user,data);
		}
	}

	public boolean isViCoordinator(User user) {
		return (coordinator.equals(user));
	}

	public void attach(User user, Session session){
		Video observer = new Video(user,session);
		Videos.add(observer);
	}

	public void notifyAllObservers(User user,String data){
		for (Video observer : Videos)
		if (!user.equals(observer.account)){
			observer.update(data);
		}
	}

	public void removeVideo(User user)
	{
		if (!user.equals(coordinator)) {
			Videos.remove(user);
		}
	}

	public boolean checkCredits(String url_link, String id){
		return (URI.equals(url_link) && meetingId.equals(id));
	}
}